pkg/uid/from_file.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2016 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// For how the uidshift and uidcount are generated please check:
+// http://cgit.freedesktop.org/systemd/systemd/commit/?id=03cfe0d51499e86b1573d1
+
+package uid
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"syscall"
+
+	"github.com/hashicorp/errwrap"
+)
+
+// FromStat is the struct that defines a uid/gid generator based on a file.
+//
+// It generates the uid/gid based on the file attribues of the given Path
+// and unshift the uid/gid if Range is non-nil.
+type FromStat struct {
```

Comment:
The target file will still be the same regardless of the invocation. A bug would be the UID or GID of a user changing in between the invocations, which I'm not sure about. I suggested it, because it's named as a generator.